backend/src/main/java/com/huellapositiva/domain/service/RemoteStorageService.java

```diff
@@ -41,7 +46,12 @@ public URL uploadProposalImage(MultipartFile image, String proposalId) throws IO
      * @throws IOException
      */
     public URL uploadVolunteerCV(MultipartFile cv, String volunteerId) throws IOException {
-        String extension = getExtension(cv.getOriginalFilename());
+        String extension;
+        try {
+            extension = getExtension(cv.getOriginalFilename());
+        } catch (NullPointerException exception) {
+            throw new FileTypeNotSupported("Curriculum vitae file must have extension (and be .pdf)");
+        }
         if(!extension.equalsIgnoreCase(".pdf")) {
```

Comment:
When comparing Strings always put the static value first to avoid NPE ".pdf".equalsIgnoreCase(extension)